### PR TITLE
make IAP optional for click-deploy app

### PR DESCRIPTION
**Pull request by kunmingg**
1. make IAP optional for click-deploy app
2. update instructions for both cases (IAP / no IAP)
3. add button for both cases to smooth experience

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/1927)

### CODE REVIEW

File: components/gcp-click-to-deploy/src/App.tsx [hunk 1]
```diff
@@ -107,22 +107,45 @@ class App extends React.Component<any, { signedIn: boolean }> {
...
+                <div>
+                  To connect to deployed Kubeflow cluster:
+                </div>
+                <div>
+                  <ul>
+                    <li> If you configured IAP Oauth Client ID and Secret: </li>
+                  </ul>
+                  <ul>
+                    <ul>
+                      <li>  Click IAP Access (might need to wait up to 20 minutes on network settings) </li>
```
Comment (jlewi) [inline components/gcp-click-to-deploy/src/App.tsx]: "20 minutes for domain and IAP to setup"

File: components/gcp-click-to-deploy/src/DeployForm.tsx [hunk 2]
```diff
@@ -262,12 +271,53 @@ export default class DeployForm extends React.Component<any, DeployFormState> {
       if (p.name === 'acmeEmail') {
         p.value = email;
       }
-    });
+
+      if (p.name === 'jupyterHubAuthenticator') {
```
Comment (jlewi) [inline components/gcp-click-to-deploy/src/DeployForm.tsx]: So the client side JS is responsible for setting the authenticator for JupyterHub?

**Comment by jlewi** [comment]
This is great; just a minor comment about the instructions.

**Comment by abhi-g** [comment]
Is it possible to add a checkbox for IAP setup, which is unchecked by default. When unchecked the IAP related fields can be hidden from the form, and visible if checked.

**Comment by abhi-g** [comment]
I agree with Jeremy that we can still merge this PR, allowing users the
option to use kubectl port-forward.

**Comment by kunmingg** [comment]
Sure, seems we might be able to automate cloud shell.
Will change frontend a bit and merge this PR.

File: components/gcp-click-to-deploy/src/App.tsx [hunk 3]
```diff
@@ -107,22 +107,45 @@ class App extends React.Component<any, { signedIn: boolean }> {
...
+                    </ul>
+                  </ul>
+                  <ul>
+                    <li> If you leave IAP Oauth Client ID and Secret empty: </li>
+                  </ul>
+                  <ul>
+                    <ul>
+                      <li>  Click Cloud Shell; run command in pop up window; kubeflow UI will be at
```
[the remark this answers is not in the document]
Comment (kunmingg) [reply components/gcp-click-to-deploy/src/App.tsx]: We provide button on UI and prepare script in user's source repo,
when user click button, cloud shell will clone repo and cd into script dir.
User only need to run ```./conn.sh```

File: components/gcp-click-to-deploy/src/DeployForm.tsx (hunk 2 above)
Comment (kunmingg) [reply components/gcp-click-to-deploy/src/DeployForm.tsx]: Ideally not, will take care in https://github.com/kubeflow/kubeflow/issues/1971

**Pull request merged**